src/test/burn/WixTestTools/RuntimePrereqFeatureFactAttribute.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved. Licensed under the Microsoft Reciprocal License. See LICENSE.TXT file in the project root for full license information.
+
+namespace WixTestTools
+{
+    using System;
+    using System.Linq;
+    using System.Management;
+    using System.Collections.Generic;
+    using System.Security.Principal;
+    using WixInternal.TestSupport.XunitExtensions;
+
+    public class RuntimePrereqFeatureFactAttribute : RuntimeFactAttribute
```

Comment:
This is cool but I don't think the functionality to install pre-reqs should be done via the `[Fact]` attribute. It doesn't need to run per-Fact. Instead, it could run once as part of a fixture configuration and be used by all the tests. So, consider moving the functionality to WixInternal.TestSupport and we may need to add the "runtime test check" (env var check) there as well so it can be disabled the same way.

Reply:
This does not install.
It only tests for the presence of the feature in the 'installed' state (I was too lazy to add Status check also, the compile cycle on WiX is brutal).

I just wanted to be able to have Facts which could exist in repo, and would be assessable, but didn't report as Failures if the scenario wasn't right (i.e. in most situations.. like possibly runners, and even the Windows Sandbox in this situation).

I think conditional Windows Feature installation from a Test Fixture is a good idea, but agree it would need to be specifically checked for by an Environment Variable check.

I think 'the shape' of the parameters here (a list of strings) is tricky to group into the existing RuntimeFactAttribute (using 'params' to avoid needing to explicitly declare the 'new string[]').  So i'd rather this one be separate (although happy for a different name), even if the Domain test can be 'pulled into' the existing RuntimeFactAttribute.

Reply:
Yeah, I didn't realize how ugly `string[]` array looked when used in as a property on an `Attribute`.  All the `static`s should tackle the initialization time.